store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java

```diff
@@ -0,0 +1,335 @@
+import uk.gov.gchq.gaffer.parquetstore.serialisation.ParquetSerialisationFactory;
+import uk.gov.gchq.gaffer.parquetstore.utils.Constants;
+import uk.gov.gchq.gaffer.parquetstore.utils.SchemaUtils;
+import uk.gov.gchq.gaffer.serialisation.Serialiser;
+import uk.gov.gchq.gaffer.spark.SparkUser;
+import uk.gov.gchq.gaffer.spark.operation.dataframe.GetDataFrameOfElements;
+import uk.gov.gchq.gaffer.spark.operation.scalardd.ImportRDDOfElements;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.StoreTrait;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.operation.handler.OutputOperationHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.store.schema.SchemaOptimiser;
+import uk.gov.gchq.gaffer.user.User;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *
+ */
+public class ParquetStore extends Store {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ParquetStore.class);
+    private static final Set<StoreTrait> TRAITS = new HashSet<>();
+    private HashMap<String, ArrayList<Tuple3<Object[], Object[], String>>> groupToIndex;
+    private SchemaUtils schemaUtils;
+    private FileSystem fs;
+    private long currentSnapshot;
+
+    static {
+        TRAITS.add(StoreTrait.INGEST_AGGREGATION);
+        TRAITS.add(StoreTrait.ORDERED);
+        TRAITS.add(StoreTrait.PRE_AGGREGATION_FILTERING);
+    }
+
+    public ParquetStore(final Schema schema, final StoreProperties properties) throws StoreException {
+        super();
+        initialise(schema, properties);
+    }
+
+    public ParquetStore() {
+        super();
+    }
+
+    @Override
+    protected SchemaOptimiser createSchemaOptimiser() {
+        return new SchemaOptimiser(new ParquetSerialisationFactory());
+    }
+
+    @Override
+    public void initialise(final Schema schema, final StoreProperties properties) throws StoreException {
+        super.initialise(schema, properties);
+        this.schemaUtils = new SchemaUtils(this.getSchema());
+        loadIndices();
+    }
+
+    public FileSystem getFS() throws StoreException {
+        if (this.fs == null) {
+            try {
+                this.fs = FileSystem.get(new Configuration());
+            } catch (IOException e) {
+                throw new StoreException("Could not connect to the file system", e);
+            }
+        }
+        return this.fs;
+    }
+
+    public SchemaUtils getSchemaUtils() {
+        if (this.schemaUtils == null) {
+            this.schemaUtils = new SchemaUtils(getSchema());
```

Comment:
this is initialised in the initialise method so you shouldn't need to do the null check